src/main/java/teammates/ui/webapi/action/GetFeedbackSessionAction.java

```diff
@@ -79,4 +82,19 @@ public ActionResult execute() {
 
         return new JsonResult(response);
     }
+
+    /**
+     * Checks the access control for instructor feedback submission.
+     */
+    private void checkInstructorForSessionAccessPrivilege(FeedbackSessionAttributes feedbackSession) {
+        String moderatedPerson = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_MODERATED_PERSON);
```

Comment:
Why there is a moderatedPerson here?

Reply:
This method is supposed to validate any access with intent `INSTRUCTOR_RESULT` which includes access to this action with non null `FEEDBACK_SESSION_MODERATED_PERSON` param. See: https://github.com/TEAMMATES/teammates/blob/bc23cd911d58bd8677dd7a70b01002d25d5bf832/src/test/java/teammates/test/cases/webapi/GetFeedbackSessionActionTest.java#L215-L221

Reply:
That's precisely because of the misconfiguration. Moderated person makes no sense when we're talking about instructor result page.

Reply:
I've pushed a refactoring of the access control mechanisms. Please help to check if I understood the nature of the misconfiguration.